From 1177e1fcaf8720ddd284dd5c74904e02a152246f Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:36 -0400
Subject: [fs] btrfs: use late_initcall instead of module_init

Message-id: <1407179109-13714-150-git-send-email-sandeen@redhat.com>
Patchwork-id: 87101
O-Subject: [RHEL7.1 PATCH 149/422] Btrfs: use late_initcall instead of module_init
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit 60efa5eb2e886852a0d5f9e1ffa7c896a1099da8
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Sat Feb 1 21:27:56 2014 +0000

    Btrfs: use late_initcall instead of module_init

    It seems that when init_btrfs_fs() is called, crc32c/crc32c-intel might
    not always be already initialized, which results in the call to crypto_alloc_shash()
    returning -ENOENT, as experienced by Ahmet who reported this.

    Therefore make sure init_btrfs_fs() is called after crc32c is initialized (which
    is at initialization level 6, module_init), by using late_initcall (which is at
    initialization level 7) instead of module_init for btrfs.

    Reported-and-Tested-by: Ahmet Inan <ainan@mathematik.uni-freiburg.de>
    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index c02f633..97cc241 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -1996,7 +1996,7 @@ static void __exit exit_btrfs_fs(void)
  btrfs_hash_exit();
 }
 
-module_init(init_btrfs_fs)
+late_initcall(init_btrfs_fs);
 module_exit(exit_btrfs_fs)
 
 MODULE_LICENSE("GPL");
-- 
1.7.1